smack-experimental/src/main/java/org/jivesoftware/smackx/sid/element/OriginIdElement.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.sid.element;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+import org.jivesoftware.smackx.sid.StableUniqueStanzaIdManager;
+
+public class OriginIdElement implements ExtensionElement {
```

Comment:
There should be a common superclass of `OriginIdElement` and `StanzaIdElement`, e.g. `UniqueAndStableIdElement`.